optuna/terminator/terminator.py

```diff
@@ -0,0 +1,37 @@
+import abc
+import math
+from typing import Optional
+
+from optuna.study.study import Study
+from optuna.terminator.improvement.evaluator import BaseImprovementEvaluator
+from optuna.terminator.improvement.evaluator import RegretBoundEvaluator
+from optuna.terminator.serror import BaseStatisticalErrorEvaluator
+from optuna.terminator.serror import CrossValidationStatisticalErrorEvaluator
+
+
+class BaseTerminator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def should_terminate(self, study: Study) -> bool:
+        pass
+
+
+class Terminator(BaseTerminator):
```

Comment:
I'm also concerned about the same point.

For example, instead of terminating based on cross-validation scores, we could add methods to see if the objective value is improved at a specific interval. In that case, this class name may not be appropriate and `CrossValidationTerminator` may be better.

Reply:
Though the name, `Terminator`, is too general, I'd like to adopt it for the following reasons.
- this would be easier for the user to remember
- we can rename it later, since this API is still experimental

Reply:
I think that's reasonable, also given we cannot come up with better alternatives that are not too verbose. Let me resolve this comment for now although it's good for record purpose.